Makefile

```diff
@@ -14,7 +14,7 @@ ALL_FEATURES_BUT_ASYNC=--features concurrent,testing
 
 .PHONY: clippy
 clippy: ## Runs Clippy with configs
-	cargo clippy --workspace --all-targets $(ALL_FEATURES_BUT_ASYNC) -- -D warnings
+	cargo clippy --workspace --all-targets $(ALL_FEATURES_BUT_ASYNC) -- -D warnings --allow clippy::arc_with_non_send_sync
```

Comment:
Had to add this because otherwise clippy complains about wrapping structs in `Arc` that are not `Send` or `Sync`, not sure how we'd want to proceed in this case

Reply:
Could this cause problems? Maybe we should try to make these structs `Send` and `Sync` (in a different PR)?

Reply:
Yes, we probably want to make these structs `Send` and `Sync`. The alternative is to move back to `Rc`.